### DISCUSSION TITLE
Possibility to add a draft tweet

### DISCUSSION THREAD

**Discussion by NicoLivesey** [tweepy/tweepy#1488] [Questions] [answered]
Hi, first thank you for your work it has been really helpful for my project.

A little bit of context, I want to build a bot based on AI generated text but I want to be the final filter who can choose whether a tweet is published or not. To do this I was thinking that the twitter draft tweets were the perfect solution. Once per day, my service would generate tweets in my drafts and I could choose whenever I want which one I publish or not.

My question now is that I don't see any mention of the possibility to add a draft tweet in your documentation while it is mentioned in the Twitter API documentation https://developer.twitter.com/en/docs/twitter-ads-api/creatives/api-reference/draft-tweets

Do you confirm that it is not a supported service for the moment ?

**Comment by Harmon758** [comment, marked as answer]
As the documentation says and as the name implies, the Twitter Ads API is used for advertising by companies.
It isn't applicable to normal user accounts, and Tweepy does not support it.
See https://developer.twitter.com/en/docs/twitter-ads-api and https://twittercommunity.com/t/announcement-tweet-drafts/106054.

Drafting Tweets isn't possible with the standard API.
See https://twittercommunity.com/t/draft-tweet-api/140162.

**Reply by NicoLivesey** [reply]
Oh yes my bad on this I didn't see I ended up in the Ads API. Do you think of a possible solution for my use case ?

Anyway thanks for the quick reply :)

**Reply by Harmon758** [reply]
As that forum post suggests, you could store and review drafts of Tweets in your own app interface.

No problem. I already converted this into a discussion, since this is a question rather than an issue with Tweepy.
